llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -433,6 +433,10 @@ void llvm::computeKnownBitsFromRangeMetadata(const MDNode &Ranges,
     // The first CommonPrefixBits of all values in Range are equal.
     unsigned CommonPrefixBits =
         (Range.getUnsignedMax() ^ Range.getUnsignedMin()).countl_zero();
+    // BitWidth must equal the Ranges BitWidth for the correct number of high
+    // bits to be set.
+    assert(BitWidth == Lower->getBitWidth() &&
+           "Known bit width must match range bit width!");
```

Comment:
This case should be caught by the IR verifier already, it doesn't need an extra assert in other code

Reply:
This assert is needed to guard against callers of computeKnownBitsFromRangeMetadata() passing in an incorrectly sized KnownBits.  Prior to https://github.com/llvm/llvm-project/pull/128908, a KnownBits with a width of a whole vector could be passed in which was a mismatch with the vector element sized range data.

Reply:
Move this up and query from Range.getBitWidth()?

Reply:
Changed to Range.getBitWidth()